experimental/web/generate_web_metrics.sh

```diff
@@ -0,0 +1,200 @@
+#
+#   then open http://localhost:8000/benchmarks.html (for automated benchmarks)
+#   or open http://localhost:8000/ (for interactive benchmarks)
+
+set -e
+
+TARGET_DIR="$1"
+if [ -z "$TARGET_DIR" ]; then
+  echo "ERROR: Expected target directory (e.g. /tmp/iree/web_metrics/)"
+  exit 1
+fi
+
+# If used in conjunction with sample_dynamic/build_sample.sh, this script can
+# copy compiled programs into that sample's build directory for benchmarking.
+ROOT_DIR=$(git rev-parse --show-toplevel)
+BUILD_DIR=${ROOT_DIR?}/build-emscripten
+SAMPLE_BINARY_DIR=${BUILD_DIR}/experimental/web/sample_dynamic/
+
+echo "Working in directory '$TARGET_DIR'"
+mkdir -p ${TARGET_DIR}
+cd ${TARGET_DIR}
+
+# Print all commands and the UTC time.
+# set -xeuo pipefail
+# export PS4='[$(date -u "+%T %Z")] '
+
+###############################################################################
+# Set up Python virtual environment                                           #
+###############################################################################
+
+python -m venv .venv
+source .venv/bin/activate
+trap "deactivate 2> /dev/null" EXIT
+
+# Skip package installs when you want by commenting this out. Freezing to a
+# specific version when iterating on metrics is useful, and fetching is slow.
+
+python -m pip install --upgrade \
+  --find-links https://github.com/google/iree/releases \
+  iree-compiler iree-tools-tflite iree-tools-xla
+
+###############################################################################
+# Download program source files                                               #
+###############################################################################
+
+wget -nc https://storage.googleapis.com/iree-model-artifacts/mobile_ssd_v2_float_coco.tflite
+wget -nc https://storage.googleapis.com/iree-model-artifacts/deeplabv3.tflite
+wget -nc https://storage.googleapis.com/iree-model-artifacts/posenet.tflite
+wget -nc https://storage.googleapis.com/iree-model-artifacts/mobilebert-baseline-tf2-float.tflite
+wget -nc https://storage.googleapis.com/iree-model-artifacts/mobilenet_v2_1.0_224.tflite
+wget -nc https://storage.googleapis.com/iree-model-artifacts/MobileNetV3SmallStaticBatch.tflite
+
+###############################################################################
+# Import programs into MLIR                                                   #
+###############################################################################
+
+# Note: you can also download imported programs from runs of the
+# https://buildkite.com/iree/iree-benchmark-android pipeline.
+
+IREE_IMPORT_TFLITE_PATH=iree-import-tflite
+
+# import_program helper
+#   Args: program_name, tflite_source_path
+#   Imports tflite_source_path to program_name.tflite.mlir
+function import_program {
+  OUTPUT_FILE=./$1.tflite.mlir
+  echo "Importing '$1' to '${OUTPUT_FILE}'..."
+  ${IREE_IMPORT_TFLITE_PATH?} $2 -o ${OUTPUT_FILE}
+}
+
+import_program "deeplabv3" "deeplabv3.tflite"
```

Comment:
The order of these arguments seems a bit wonky to me. It made me think that the function was outputting .tflite files

Reply:
Hm :/ it makes sense to me, but this could really benefit from some named arguments or other more advanced script features :P. Probably worth moving to Python if this gets more complex.